src/TurbulenceConvection/EDMF_functions.jl

```diff
@@ -766,26 +767,18 @@ function compute_up_tendencies!(
         (; bottom = CCO.SetValue(wvec(FT(0))), top = CCO.SetValue(wvec(FT(0))))
     LBC = CCO.LeftBiasedF2C(; bottom = CCO.SetValue(FT(0)))
     ∇f = CCO.DivergenceC2F(; adv_bcs...)
-
     @inbounds for i in 1:N_up
-        a_up_bcs = a_up_boundary_conditions(surf, edmf, i)
-        Iaf = CCO.InterpolateC2F(; a_up_bcs...)
-        ρaw = prog_up_f[i].ρaw
-        tends_ρaw = tendencies_up_f[i].ρaw
+        w = prog_up_f[i].w
+        tends_w = tendencies_up_f[i].w
         nh_pressure = aux_up_f[i].nh_pressure
-        a_up = aux_up[i].area
-        w_up = aux_up_f[i].w
+        w_up = prog_up_f[i].w
         w_en = aux_en_f.w
         entr_w = aux_up[i].entr_turb_dyn
-        detr_w = aux_up[i].detr_turb_dyn
         buoy = aux_up[i].buoy
 
-        @. tends_ρaw = -(∇f(wvec(LBC(ρaw * w_up))))
-        @. tends_ρaw +=
-            (ρaw * (I0f(entr_w) * w_en - I0f(detr_w) * w_up)) +
-            (ρ_f * Iaf(a_up) * I0f(buoy)) +
-            nh_pressure
-        tends_ρaw[kf_surf] = 0
+        @. tends_w = -w * (∇f(wvec(LBC(w))))
+        @. tends_w += w * I0f(entr_w) * (w_en - w) + I0f(buoy) + nh_pressure
```

Comment:
Looks like there's 1/(a_up rho) missing in front of the pressure gradient term?

Reply:
I removed the rho*a that was multiplying the pressure gradient term in the previous code. there is need to multiply and than divide by the same quantity